tests/integration_tests/test_wave_04_05_06_07.py

```diff
@@ -0,0 +1,115 @@
+import pytest
+from swap_meet.vendor import Vendor
+from swap_meet.clothing import Clothing
+from swap_meet.decor import Decor
+from swap_meet.electronics import Electronics
+
+@pytest.mark.skip
+@pytest.mark.integration_test
+def test_integration_wave_04_05_06(capfd):
+    camila = Vendor()
+    valentina = Vendor()
+
+    item_clothing1 = Clothing(condition=1.0, id=123, fabric="Geometric Pattern")
+    item_clothing2 = Clothing(condition=2.0, id=321)
+    item_electronics1 = Electronics(condition=1.0, id=456)
+    item_electronics2 = Electronics(condition=2.0, id=654, type="Kitchen Appliance")
+    item_decor1 = Decor(condition=1.0, id=789)
+    item_decor2 = Decor(condition=2.0, id=987, width=4, length=2)
+
+    camila.add(item_electronics1)
+    camila.add(item_clothing1)
+    camila.add(item_clothing2)
+
+    valentina.add(item_electronics2)
+    valentina.add(item_decor1)
+    valentina.add(item_decor2)
+
+    # swap first item
+    result = camila.swap_first_item(valentina)
+    
+    assert result
+    assert len(camila.inventory) == 3
+    assert item_clothing1 in camila.inventory
+    assert item_clothing2 in camila.inventory
+    assert item_electronics2 in camila.inventory
+
+    assert len(valentina.inventory) == 3
+    assert item_decor1 in valentina.inventory
+    assert item_decor2 in valentina.inventory
+    assert item_electronics1 in valentina.inventory
+
+    # get item by category, truthy
+    items = camila.get_by_category("Electronics")
+    assert len(items) == 1
+    assert item_electronics2 in items
+
+    # get item by category, falsy
+    items = valentina.get_by_category("Clothing")
+    assert len(items) == 0
```

Comment:
This either needs to be changed, or we need to change the requirement in the `get_by_category` wave to return an empty list instead of `None`. In some way an empty list makes more sense (this is feedback that Ansel gave on the higher order functions lesson), but I think either is okay.

```suggestion
    assert items == None
```

Reply:
Ooo, thank for catching this! Since `get_by_category` is filtering another list, it makes a lot of sense to return an empty list rather than `None`, I'll make those updates.